### PR TITLE
Add support for redis cluster

### PR DESCRIPTION
**Pull request by labbati**
### Description

This PR adds support for redis cluster operations to the [phpredis](https://github.com/phpredis/phpredis) insturmentation.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [x] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

**Review by SammyK — COMMENTED** [review]
Nice work on adding `RedisCluster` @labbati. 💯 I left a few small suggestions but the main suggestion is adding PHP 8 support.

File: ext/php7/handlers_phpredis.c [hunk 1]
```diff
@@ -210,6 +211,8 @@ void ddtrace_phpredis_handlers_startup(void) {
     // clang-format on
 
     ddtrace_string phpredis = DDTRACE_STRING_LITERAL("redis");
+    ddtrace_string phpredis_cluster = DDTRACE_STRING_LITERAL("rediscluster");
     size_t methods_len = sizeof methods / sizeof methods[0];
     ddtrace_replace_internal_methods(phpredis, methods_len, methods);
+    ddtrace_replace_internal_methods(phpredis_cluster, methods_len, methods);
```
[the remark this answers is not in the document]
Comment (labbati) [reply ext/php7/handlers_phpredis.c:217]: That confused me as well when I went there :) in PHP8 we load `predis`, not `phpredis` which is [executed](https://github.com/DataDog/dd-trace-php/blob/8ecc1ce70df81cd837357ed80d7514175d30c304/Makefile#L520).

[the remark this answers is not in the document]
Comment (labbati) [reply ext/php7/handlers_phpredis.c:217]: Still, you are right that we load but not execute them (probably a left over of the directory cloning?). Let me run locally to confirm everything works.

**Review by SammyK — COMMENTED** [review]
Thanks for addressing the PHP 8 issue, @labbati! 👍
Somehow the Slim v4 changes got added to this PR?

File: ext/php7/handlers_phpredis.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (SammyK) [reply ext/php7/handlers_phpredis.c:217]: Ah, `predis`. Sorry I missed that. Thanks for digging into the PHP 8 stuff!

**Review by SammyK — APPROVED** [review]
Nice work @labbati!

**Pull request merged**